core/common/boot/src/main/java/org/eclipse/edc/boot/system/DependencyGraph.java

```diff
@@ -121,6 +130,22 @@ public List<InjectionContainer<ServiceExtension>> of(List<ServiceExtension> load
                 .collect(Collectors.toList());
     }
 
+    private boolean canResolve(Map<String, List<ServiceExtension>> dependencyMap, String featureName) {
```

Comment:
I already approved and I'm ok with the whole PR, but I noticed that `InjectionPoint.featureName` is always the same as `InjectionPoint.injectedField().getType().getName()` (see the only constructor call in `InjectionPointScanner`), so here the injection point could be called so the `Class.forName` at line 140 can be removed and replaced with `injectionPoint.getType()`

Reply:
yep, makes sense.